emscripten.py

```diff
@@ -471,7 +471,7 @@ def make_emulated_param(i):
                                 'shiftLeftByScalar'];
     fundamentals = ['Math']
     if settings['USE_PTHREADS']:
-      fundamentals += ['SharedInt8Array', 'SharedInt16Array', 'SharedInt32Array', 'SharedUint8Array', 'SharedUint16Array', 'SharedUint32Array', 'SharedFloat32Array', 'SharedFloat64Array', 'Atomics']
+      fundamentals += ['I8Array', 'I16Array', 'I32Array', 'U8Array', 'U16Array', 'U32Array', 'F32Array', 'F64Array', 'Atomics']
```

Comment:
I was thinking about that as well, but I leaned towards this form to be sharp to not disrupt the codegen for the non-threads case and follow "don't pay what you don't use". If I used the `I8Array` etc. in both, then non-threads builds would change to the form

	var I8Array = Int8Array;
	var I16Array = Int16Array;
	var I32Array = Int32Array;
	var U8Array = Uint8Array;
	var U16Array = Uint16Array;
	var U32Array = Uint32Array;
	var F32Array = Float32Array;
	var F64Array = Float64Array;

	Module.asmGlobalArg = { "I8Array": I8Array, "I16Array": I16Array, "I32Array": I32Array, "U8Array": U8Array, "U16Array": U16Array, "U32Array": U32Array, "F32Array": F32Array, "F64Array": F64Array };
	var asm = (function(global, env, buffer) {
	  var HEAP8 = new global.I8Array(buffer);
	  var HEAP16 = new global.I16Array(buffer);
	  var HEAP32 = new global.I32Array(buffer);
	  var HEAPU8 = new global.U8Array(buffer);
	  var HEAPU16 = new global.U16Array(buffer);
	  var HEAPU32 = new global.U32Array(buffer);
	  var HEAPF32 = new global.F32Array(buffer);
	  var HEAPF64 = new global.F64Array(buffer);
	  ...
	}

which looks a bit wasteful, and it seems better to have the complexity in the compiler side instead of the generated code side, so that no extra characters are introduced to non-threads builds.